plugins/languages/cmake-tools.nix

```diff
@@ -0,0 +1,134 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "cmake-tools";
+  originalName = "cmake-tools.nvim";
+  defaultPackage = pkgs.vimPlugins.cmake-tools-nvim;
+
+  maintainers = [ helpers.maintainers.NathanFelber ];
+
+  extraPlugins = with pkgs.vimPlugins; [ plenary-nvim ];
+
```

Comment:
Could you please add a `settingsExample` section as well ?

Reply:
Note for context: I'd previously suggested removing the example as it just contained the plugin defaults.

If we can come up with a useful non-default example that'd be ideal though.

It could be extensive, or it could just set one or two "important" or frequently configured options.

Maybe try searching `lang:lua "cmake-tools"` accross GitHub to see real world examples, or check if upstream has any examples in its docs?

Reply:
Sure, I'll add one with non-default values.

Reply:
I added a basic example. It only slightly deviates from the default configuration, but in what I consider could be a useful (or at least usable) way. I wasn't able to find examples online or upstream unfortunately, but the `cmake_dap_configuration` part comes from [the nvim-dap wiki](https://github.com/mfussenegger/nvim-dap/wiki/C-C---Rust-(via--codelldb)).